### PR TITLE
[FLINK-7535] Port DashboardConfigHandler to new REST endpoint

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

Lets DashboardConfigHandler implement the LegacyRestHandler. Moreover, this
commit defines the appropriate DashboardConfigurationHeaders.

The DispatcherRestEndpoint registers the DashboardConfigHandler.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/messages/DashboardConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	@JsonProperty(FIELD_NAME_TIMEZONE_NAME)
+	private final String timeZoneName;
+
+	@JsonProperty(FIELD_NAME_TIMEZONE_OFFSET)
+	private final int timeZoneOffset;
+
+	@JsonProperty(FIELD_NAME_FLINK_VERSION)
+	private final String flinkVersion;
+
+	private final String flinkRevision;
```
Comment (tzulitai) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/messages/DashboardConfiguration.java]: this field is missing the Jackson annotation.

Comment (tzulitai) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/messages/DashboardConfiguration.java]: Which means we would want a Json marshalling test for this also.

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/RestHandlerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+public class RestHandlerConfiguration {
+
+	private final long refreshInterval;
+
+	private final Time timeout;
+
+	private final File tmpDir;
+
+	public RestHandlerConfiguration(long refreshInterval, Time timeout, File tmpDir) {
+		this.refreshInterval = refreshInterval;
```
Comment (tzulitai) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/RestHandlerConfiguration.java]: Is it required that we have a validation check for this? Although I don't think we ever had a validation check for this before.

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/messages/DashboardConfiguration.java (hunk 1 above)
Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/messages/DashboardConfiguration.java]: Good catch. I'll add the marshalling test.

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/RestHandlerConfiguration.java (hunk 2 above)
Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/RestHandlerConfiguration.java]: Adding a sanity check wouldn't hurt. I'll add it.

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/DashboardConfigurationHeaders.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import org.apache.flink.runtime.rest.HttpMethodWrapper;
+import org.apache.flink.runtime.rest.handler.legacy.DashboardConfigHandler;
+import org.apache.flink.runtime.rest.handler.legacy.messages.DashboardConfiguration;
+
+import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpResponseStatus;
+
+/**
+ * Message headers for the {@link DashboardConfigHandler}.
+ */
+public class DashboardConfigurationHeaders implements MessageHeaders<EmptyRequestBody, DashboardConfiguration, EmptyMessageParameters> {
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/DashboardConfigurationHeaders.java]: This could be a singleton

Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/DashboardConfigurationHeaders.java]: True. The same applies to `ClusterOverviewHeaders`. Will change it.

**Comment by tillrohrmann** [comment]
Thanks for the review @zentol and @tzulitai. Merging this PR now.

**Pull request closed without merging**